appserver/payara-appserver-modules/microprofile/jwt-auth/src/main/java/fish/payara/microprofile/jwtauth/eesecurity/SignedJWTIdentityStore.java

```diff
@@ -217,10 +218,21 @@ public CredentialValidationResult validate(SignedJWTCredential signedJWTCredenti
             return Optional.empty();
         }
 
-        byte[] byteBuffer = new byte[16384];
-        try (InputStream inputStream = publicKeyURL.openStream()) {
-            String key = new String(byteBuffer, 0, inputStream.read(byteBuffer));
-            return createPublicKey(key, keyID);
+        URLConnection urlConnection = publicKeyURL.openConnection();
+        Charset charset = Charset.defaultCharset();
+        ContentType contentType = ContentType.newContentType(urlConnection.getContentType());
+        if(contentType != null) {
+            String charEncoding = contentType.getCharacterEncoding();
+            if(!Charset.isSupported(charEncoding)){
+                LOGGER.severe("Charset " + charEncoding + " for remote public key not supported, using default charset instead");
```

Comment:
This may throw exceptions if anything funny or NULL is provided.

Reply:
Thanks for the suggestions